scripts/docker/ubuntu.16.04/Dockerfile

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+FROM ubuntu:16.04
+
+# Install the base toolchain we need to build anything (clang, cmake, make and the like)
```

Comment:
What do you think about sharing between the ubuntu and the ubuntu.16.04 dockerfiles?  These seem to be really duplicated.

Maybe you can add a common bash script that has the common pieces between the 2 versions, so that we we don't need to maintain this twice.